src/store/messagesStore.js

```diff
@@ -321,6 +324,29 @@ const actions = {
 	deleteMessages(context, token) {
 		context.commit('deleteMessages', token)
 	},
+
+	async clearLastReadMessage(context, { token }) {
+		const conversation = context.getters.conversations[token]
+		if (!conversation || !conversation.lastMessage) {
+			return
+		}
+		// set the id to the last message
+		context.dispatch('updateLastReadMessage', { token, id: conversation.lastMessage.id })
+		context.dispatch('markConversationRead', token)
+	},
+
+	async updateLastReadMessage(context, { token, id = 0 }) {
+		const conversation = Object.assign({}, context.getters.conversations[token])
+		if (!conversation || conversation.lastReadMessage === id) {
+			return
+		}
+
+		// optimistic early commit to avoid indicator flickering
+		context.commit('updateConversationLastReadMessage', { token, lastReadMessage: id })
+		await updateLastReadMessage(token, id)
+
+		// TODO: update unread counters if possible (otherwise user needs to wait for conversation list refresh)
```

Comment:
Should the readmarker endpoint reply with that?

Reply:
Obsolete, I'll remove. There was a bug that was hiding the fact that this already works.